File: iPhone/ScreenShot/ScreenShot.js
/*
 *  This code is adapted from the work of Michael Nachbaur 
 *  by Simon Madine of The Angry Robot Zombie Factory
 *  2010-05-04
 *  MIT licensed
*/

/**
 * This class exposes the ability to take a Screenshot to JavaScript
 * @constructor
 */
function Screenshot() {
}

/**
 * Save the screenshot to the user's Photo Library
 */
Screenshot.prototype.saveScreenshot = function( returnBase64, successCallbackString) {
    PhoneGap.exec("Screenshot.saveScreenshot", returnBase64, successCallbackString );
};

Screenshot.prototype.saveScreenshotAsFile = function( fileName, successCallbackString, returnBase64 ) {
    PhoneGap.exec("Screenshot.saveScreenshotAsFile", fileName, successCallbackString, returnBase64 );
};

PhoneGap.addConstructor(function() 
{
	if(!window.plugins)
	{
		window.plugins = {};
	}
    window.plugins.screenShot = new Screenshot();
});

File: iPhone/ScreenShot/ScreenShot.m
//
//  Screenshot.h
//
//  Created by Simon Madine on 29/04/2010.
//  Copyright 2010 The Angry Robot Zombie Factory.
//  MIT licensed
//
//  Modifications to support orientation change by @ffd8
//

#import "Screenshot.h"
@implementation Screenshot

- (void)saveScreenshot:(NSArray*)arguments withDict:(NSDictionary*)options
{

    CGRect screenRect = [[UIScreen mainScreen] bounds];
    CGRect imageRect;

    //change with orientation
    UIInterfaceOrientation orientation = [[UIDevice currentDevice] orientation];
    if (orientation == UIInterfaceOrientationPortrait || orientation == UIInterfaceOrientationPortraitUpsideDown) { 
            //portrait check
            imageRect = CGRectMake(0, 0, CGRectGetWidth(screenRect), CGRectGetHeight(screenRect));
        }else if(orientation == UIInterfaceOrientationLandscapeLeft || orientation == UIInterfaceOrientationLandscapeRight) { 
            //landscape check
            imageRect = CGRectMake(0, 0, CGRectGetHeight(screenRect), CGRectGetWidth(screenRect));
        }else{
            //default with portrait (after tests.. worked best with this code here)
            imageRect = CGRectMake(0, 0, CGRectGetWidth(screenRect), CGRectGetHeight(screenRect));
        }


    UIGraphicsBeginImageContext(imageRect.size);

    CGContextRef ctx = UIGraphicsGetCurrentContext();
    [[UIColor blackColor] set];
    CGContextTranslateCTM(ctx, 0, 0);
    CGContextFillRect(ctx, imageRect);

    [webView.layer renderInContext:ctx];

    UIImage *image1 = UIGraphicsGetImageFromCurrentImageContext();
    UIImageWriteToSavedPhotosAlbum(image1, nil, nil, nil);
    UIGraphicsEndImageContext();
    UIAlertView *alert= [[UIAlertView alloc] initWithTitle:nil message:@"Image Saved" delegate:self cancelButtonTitle:@"OK" otherButtonTitles:nil];
    [alert show];
    [alert release];
}
@end